services/brig/src/Brig/API/Internal.hs

```diff
@@ -472,16 +472,13 @@ createUserNoVerifySpar uData =
        in API.activate key code (Just uid) !>> CreateUserSparRegistrationError . activationErrorToRegisterError
     pure . SelfProfile $ usr
 
-deleteUserNoVerifyH :: UserId -> (Handler r) Response
-deleteUserNoVerifyH uid = do
-  setStatus status202 empty <$ deleteUserNoVerify uid
-
-deleteUserNoVerify :: UserId -> (Handler r) ()
-deleteUserNoVerify uid = do
-  void $
-    lift (wrapClient $ API.lookupAccount uid)
-      >>= ifNothing (errorToWai @'E.UserNotFound)
-  lift $ API.deleteUserNoVerify uid
+deleteUserNoAuthH :: UserId -> (Handler r) Response
+deleteUserNoAuthH uid = do
+  r <- lift $ wrapHttp $ API.ensureAccountDeleted uid
+  pure $ case r of
+    NoUser -> setStatus status404 $ json r
```

Comment:
Is this different from `errorToWai @'E.UserNotFound`?  I think it is, and I `r` is rendered into a top-level string, correct?  If so, does any of the call sides care?   And can we not have top-level string bodies?

To be more specific, can we keep `errorToWai @'E.UserNotFound` in the `NoUser` case and respond with empty both otherwise, and distinguish the cases based on the status code on the call side?

Reply:
The answer to most of your questions is: *Yes* :smile_cat: 

Will update the code accordingly.